### PR TITLE
FeeHandler fix

### PR DESCRIPTION
**Pull request by martinvol**
* Allows distributions without burn.
* Enables burn fraction to be zero

### CODE REVIEW

File: packages/protocol/contracts/common/FeeHandler.sol [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 pragma solidity ^0.5.13;
 
+import { console } from "forge-std/console.sol";
```
Comment (pahor167) [inline packages/protocol/contracts/common/FeeHandler.sol]: can we remove console import ?

Comment (martinvol) [reply packages/protocol/contracts/common/FeeHandler.sol]: ups

File: packages/protocol/contracts/common/FeeHandler.sol [hunk 2]
```diff
@@ -384,6 +397,10 @@ contract FeeHandler is
     return tokenStates[tokenAddress].toDistribute;
   }
 
+  function getTokenToBurn(address tokenAddress) external view returns (uint256) {
+    return tokenStates[tokenAddress].toDistribute;
```
Comment (pahor167) [inline packages/protocol/contracts/common/FeeHandler.sol]: Should this be rather toBurn (not toDistribute)?

File: packages/protocol/contracts/common/FeeHandler.sol [hunk 3]
```diff
@@ -485,21 +502,26 @@ contract FeeHandler is
...
-      .fromFixed(); //here2
+      .fromFixed();
+    tokenState.toBurn = tokenState.toBurn.add(balanceToBurn);
     tokenState.toDistribute = tokenState.toDistribute.add(balanceToProcess.sub(balanceToBurn));
-    return balanceToBurn;
+    return tokenState.toBurn;
+  }
+
+  function shouldBurn() public view returns (bool) {
+    return _getBurnFraction() > 0;
```
Comment (pahor167) [inline packages/protocol/contracts/common/FeeHandler.sol]: small gas saving: getBurnFraction() != 0;

File: packages/protocol/contracts/common/FeeHandler.sol [hunk 4]
```diff
@@ -552,17 +574,12 @@ contract FeeHandler is
...
-    uint256 balanceToProcess = balanceOfCelo.sub(tokenState.toDistribute).sub(celoToBeBurned);
-    uint256 balanceToBurn = _setDistributeAfterBurn(tokenState, balanceToProcess);
-    uint256 totalBalanceToBurn = balanceToBurn.add(celoToBeBurned);
-    celoToBeBurned = 0;
+    address celoTokenAddress = getCeloTokenAddress();
+    TokenState storage tokenState = tokenStates[celoTokenAddress];
+    _setDistributionAndBurnAmounts(tokenState, IERC20(celoTokenAddress));
 
-    celo.burn(totalBalanceToBurn);
+    ICeloToken(celoTokenAddress).burn(tokenState.toBurn);
```
Comment (pahor167) [inline packages/protocol/contracts/common/FeeHandler.sol:581]: should we check if toBurn != 0 ?

File: packages/protocol/contracts/common/FeeHandler.sol (hunk 2 above)
Comment (martinvol) [reply packages/protocol/contracts/common/FeeHandler.sol]: very true

**Review by pahor167 — APPROVED** [review]

**Pull request merged**